Fix Edible Blob bugs

Random-Scientist opened the pull request:
As noted in #105 , the current Edible Blob implementation is quite buggy since most Vanilla/Forge methods that involve foods work with `FoodProperties`', which are never provided by the Edible Blob code since it uses a custom, NBT-based food system that never touches Vanilla code. My implementation takes advantage of the fact that Forge allows you to override the `getFoodProperties()` method in `IForgeItem` to return a vanilla/mod friendly `FoodProperties` that is built on-the-fly with hunger and saturation values from the `ItemStack` of blobs (as well as an identifier that allows you to feed dogs with them). This along with a simple override of the `isEdible()` method (the default method would always return false since our `FoodProperties` aren't hardcoded into the item) allows for improved mod compatibility, fixes the bugs when feeding dogs  all while simplifying the code drastically (since we can remove all of the custom hunger management code and allow Vanilla to do it for us)

Edit: The code in this PR will throw an NPE if the "food" or "sat" NBT tags are missing, but that can easily be fixed with a couple of null checks, and NBT-less edible blobs are unobtainable without commands anyways.

Random-Scientist committed:
Fix EdibleBlob

Da-Technomancer commented:
The saturation values for this PR is incorrect. 
FoodProperties uses a different (float-based) definition of 'saturation' from the NBT value (int) on the edible blob.
The int in nbt is the total points of saturation restored, full stop.
The FoodProperties float definition of saturation is 'points of saturation restored as a fraction of hunger. So a saturation of 1 with hunger 9 would actually restore 9 points of saturation. Because you don't convert the value before setting it as saturation, (for example) a hunger 8, saturation 4 blob would actually restore _32 saturation_ (4F * 8). 
This float definition of saturation doesn't support 0 hunger, non-zero saturation (because you'd need an infinite saturation modifier); this could be fixed by forcing edible blocks to restore at least 1 point of hunger.

Random-Scientist committed:
limit all new fat blobs to have >=1 hunger and force old 0 hunger blobs to have one.

Random-Scientist committed:
update docs to reflect Fat Congealer and Edible Blob changes

Random-Scientist commented:
Fixed the issue with incorrect saturation and added documentation for the necessary changes to the Fat Congealer (new Blobs cannot have 0 hunger)

Da-Technomancer committed:
Backwards compat fix
*Fixed tooltips on older edible blobs (which could have 0 hunger) still showing them as 0 hunger despite now restoring 1
